src/components/Dropzone/Dropzone.vue

```diff
@@ -0,0 +1,345 @@
+          </span>
+          <span v-if="fileTypesArray.length > 1">
+            {{ textContent.acceptedFormatsAre }} <strong>{{ fileTypesString }}</strong>.
+          </span>
+        </span>
+        <span v-if="maxSizeInBytes">
+          {{ textContent.maxFileSizeIs }} <strong>{{ formatBytes(maxSizeInBytes) }}</strong>.
+        </span>
+      </div>
+      <LobLink
+        v-if="sampleLinkUrl"
+        :to="sampleLinkUrl"
+        target="_blank"
+        class="!text-sm"
+      >
+        {{ textContent.downloadSampleFile }}
+      </LobLink>
+    </div>
+  </div>
+</template>
+
+<script>
+import { formatBytes } from '@/utils/formatBytes';
+import { LobLink, ProgressBar, Upload } from '@/components';
+
+export default {
+  name: 'Dropzone',
+  components: { LobLink, ProgressBar, Upload },
+  props: {
+    textContent: { type: Object, required: true,
+      validator: function (value) {
+        const requiredKeys = ['dragAndDropHere', 'yourFile', 'or', 'uploadFileButtonText',
+          'acceptedFormatIs', 'acceptedFormatsAre', 'maxFileSizeIs', 'downloadSampleFile',
+          'uploading', 'mightTakeAMinute', 'looksGreat', 'removeFileButtonText',
+          'couldNotUpload', 'canOnlySelectOneFile', 'fileIsTooLarge', 'fileTypeNotValid',
+          'defaultErrorText', 'errorMessage', 'successMessage'];
+        for (const key of requiredKeys) {
+          return (Object.keys(value).includes(key));
+        }
+      }
+    },
+    uploadId: { type: String, required: true },
+    acceptType: { type: String, default: '/*' },
+    maxSizeInBytes: { type: [Number, null], default: null },
+    showTypeAndMaxSize: { type: Boolean, default: true },
+    sampleLinkUrl: { type: String, default: null },
+    progress: { type: [Number, null], default: null },
+    fileOb: {  type: Object, default: null },
+    status: { type: [String, null], default: null,
+      validator: function (value) {
+        return [null, 'error', 'success'].includes(value);
+      }
+    }
+  },
+  emits: ['select', 'remove'],
+  data: function () {
+    return {
+      formatBytes,
+      selectedFile: this.fileOb,
+      currentStep: this.status !== null ? this.status :  'default',
+      dragOverlay: false,
+      multipleFilesError: false,
+      fileSizeError: false,
+      fileTypeError: false
+    };
+  },
+  computed: {
+    defaultStep () {
+      return this.currentStep === 'default';
+    },
+    errorStep () {
+      return this.currentStep === 'error';
+    },
+    uploadingStep () {
+      return this.currentStep === 'uploading';
+    },
+    successStep () {
+      return this.currentStep === 'success';
+    },
+    fileHasBeenSelected () {
+      return this.successStep && this.selectedFile;
+    },
+    selectedFileType () {
+      return this.selectedFile.type ? this.selectedFile.type.split('/')[1].toUpperCase() : 'File';
+    },
+    selectedFileName () {
+      return this.selectedFile.name ? this.selectedFile.name : 'File';
```

Comment:
```suggestion
      return this.selectedFile.name || 'File';
```

Reply:
Though I wonder if the fallback value should be coming from `textContent`, maybe as `defaultFileName` or something like that?

Reply:
this is actually from the initial versions of the fileUploader, we may not need it? is there any chance of a file not having a name? I remember discussing that it may not always have a 'type', but it should always have a name, right?

Reply:
Good question.  I don't know that off the top of my head but that would be good to look into.  I'm guessing it would always have a name but we probably should be sure.

Reply:
The way I am reading this, name and size are always there, but type may be an empty string [https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications](https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications)

Reply:
so keeping `this.selectedFile.name` should be enough

Reply:
Nice, thanks for looking into that 🙌